### PR TITLE
Port-sniffer base structure

### PR DESCRIPTION
**Pull request by 0to1to0to1**
`Add port-sniffer by svart0222`

I've read `./README.md` and `./CODE_QUALITY.md` carefully.

The code is checked by `yarn run lint:js` and linter reported no errors. 

The code is submitted in its own sub-directory and in a dedicated feature branch.

Please, review.

### CODE REVIEW

File: submissions/svart0222/port-sniffer/index.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    });
+  });
+};
+
+const getResult = async (host, startPoint = 0, endPoint = 65535) => {
+  try {
+    const result = [];
+    let currentPoint = startPoint;
+
+    while (currentPoint < endPoint) {
```
Comment (denysovkos) [inline submissions/svart0222/port-sniffer/index.js:26]: Try to replace while loop, on for loop, and use break and continue statements. 
While it is a blocking operation, and not required in this case. 

Also, checkPort function returns a promise, that should be resolved, you need to resolve it somehow, and push to results only actual results

File: submissions/svart0222/port-sniffer/index.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+      result.push(checkPort(currentPoint, host));
+      currentPoint += 1;
+    }
+    const scanResult = await Promise.all(result);
```
Comment (denysovkos) [inline submissions/svart0222/port-sniffer/index.js:30]: Look at comment bellow.
This looks like:
1. [....a lot of promises can be here]
2. lest resolve everything there

What will happen, if the number of operations in your call stack will be > 65,3k?

That's why, if you are using iterations approach, let's do async calls like they are sync.

File: submissions/svart0222/port-sniffer/index.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    scanResult.forEach(el => {
+      if (el.result) process.stdout.write(`Port ${el.port} is opened\n`);
+    });
+  } catch (e) {
+    process.stdout.write(e);
```
Comment (denysovkos) [inline submissions/svart0222/port-sniffer/index.js:35]: better use console.log

File: submissions/svart0222/port-sniffer/index.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    const result = [];
+    let currentPoint = startPoint;
+
+    while (currentPoint < endPoint) {
+      result.push(checkPort(currentPoint, host));
+      currentPoint += 1;
+    }
+    const scanResult = await Promise.all(result);
+    scanResult.forEach(el => {
+      if (el.result) process.stdout.write(`Port ${el.port} is opened\n`);
```
Comment (denysovkos) [inline submissions/svart0222/port-sniffer/index.js:32]: Here, better use console.table

File: submissions/svart0222/port-sniffer/index.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+const startCheck = async () => {
+  try {
+    const { host, portStart, portEnd, help } = parseArgs(process.argv.slice(2));
+    if (help) {
+      process.stdout.write(help);
+      process.exit(0);
+    }
+    await getResult(host, portStart, portEnd);
+    process.exit(0);
```
Comment (denysovkos) [inline submissions/svart0222/port-sniffer/index.js:68]: unnecessary call

**Review by denysovkos — CHANGES_REQUESTED** [review]

**Comment by 0to1to0to1** [comment]
@denysovkos , thanks for your recommendation. 
Now I check the code and it doesn't work. I fix this bug and will make changes according to your recommendation.

File: submissions/svart0222/port-sniffer/index.js (hunk 1 above)
Comment (0to1to0to1) [reply submissions/svart0222/port-sniffer/index.js:26]: Sorry, can't understand why for loop is better than while loop ? 
if I correctly understand you recommend wait result from every check, yes ?

File: submissions/svart0222/port-sniffer/index.js (hunk 3 above)
Comment (0to1to0to1) [reply submissions/svart0222/port-sniffer/index.js:35]: but if we will use console we can't see result in the terminal (if call from terminal, without debugger), am I right ?

File: submissions/svart0222/port-sniffer/index.js (hunk 5 above)
Comment (0to1to0to1) [reply submissions/svart0222/port-sniffer/index.js:68]: Sorry, what do you mean ? Unnecessary call `process.exit(0)` or something other ?

**Pull request closed without merging**